### DISCUSSION TITLE
About SourceType

### DISCUSSION THREAD

**Discussion by FURUNZ** [SeisSol/SeisSol#1090] [Q&A] [answered]
About Sourcetypes I saw in the pointsource example (LOH) that there are two types sourcetype=42，and sourcetype=50, is there any other type of sourcetype I can find in seissol?  Can I edit the NetCDF Rupture Format implementation myself to change the type of the point source?  Thank you！

**Comment by sebwolf-de** [comment]
There are no options other than 42 and 50.
SeisSol is open source, so you can edit the implementation as you wish. If you have implemented a new feature, we are happy to receive a pull request :)

**Comment by FURUNZ** [comment]
about sourcetype 50
header=\
""" seismic moment tensor
 1.12000E+19 0.00000E+00 0.00000E+00
 0.00000E+00 0.00000E+00 0.00000E+00
 0.00000E+00 0.00000E+00 0.00000E+00
 number of faults
                     1
 x,y,z,strike, dip, rake, area, onset time.
     0.00000     0.00000     781.00000     0.00000    0.00000     0.00000  1.0     0.00000
 source time function: dela, total sample point
     0.00020                  10000
 samples
"""
Setting different moment tensor matrix will get different mechanism and magnitude of the source, what is the relationship between this M0 and magnitude, please give the reference, add I want to use different magnitude earthquake, how to set the moment tensor value

**Reply by sebwolf-de** [reply, marked as answer]
The header you posted will generate a single couple earthquake. 
The moment is in standard SI units, so in this case, you'll have 1.12E19 Pa moment. 
There is a plethora of magnitued definitions, in particular, if you consider local magnitudes. I'd suggest to start off with the definitions on wikipedia and then read the references therein: https://en.wikipedia.org/wiki/Moment_magnitude_scale#Moment_magnitude_scale. Otherwise, I suggest to read chapter 3 in the classic Aki&Richards textbook.
